lib/instrumentation/generic-span.js

```diff
@@ -67,6 +67,17 @@ Object.defineProperty(GenericSpan.prototype, 'sampled', {
   }
 })
 
+Object.defineProperty(GenericSpan.prototype, 'sampleRate', {
+  enumerable: true,
+  get () {
+    const rate = parseFloat(this._context.tracestate.getValue('s'))
+    if (rate || rate === 0) {
```

Comment:
1. Should this discard values outside of `[0, 1]` as invalid?  The *comments* in apm-server (https://github.com/elastic/apm-server/blob/b92191369f58c939ac894b478f1bb6385c07dd63/model/modeldecoder/v2/model.go#L768-L771) and the spec suggest it should be in that range, though it doesn't look to me that apm-server validates that.

2. nit: Also perhaps use `if (!isNaN(rate)) {`  to guard the result from `parseFloat()`.

Reply:
FWIW, I'm fine punting the range check to a separate issue .

Reply:
Longer term it's probably worth thinking about where/when values that could be validated should be validated. 

For now, validating this value here seems fine, as it also ends up protecting against `NaN`.

```
    if(rate >=0 || rate <= 1) {
      return rate
    }
```